controllers/openstackcontrolplane_controller.go

```diff
@@ -695,20 +591,454 @@ func (r *OpenStackControlPlaneReconciler) createOrUpdateOpenStackMACAddress(inst
+			err = common.WrapErrorForObject(cond.Message, instance, err)
+
 			return err
 		}
 
 		return nil
 	})
 	if err != nil {
-		newProvStatus.State = ospdirectorv1beta1.ControlPlaneError
-		newProvStatus.Reason = err.Error()
-		_ = r.setProvisioningStatus(instance, newProvStatus)
+		cond.Message = fmt.Sprintf("Failed to create or update OpenStackMACAddress %s ", instance.Name)
+		cond.Reason = ospdirectorv1beta1.ConditionReason(ospdirectorv1beta1.MACCondReasonCreateMACError)
+		cond.Type = ospdirectorv1beta1.ConditionType(ospdirectorv1beta1.CommonCondTypeError)
+		err = common.WrapErrorForObject(cond.Message, instance, err)
+
 		return err
 	}
 	if op != controllerutil.OperationResultNone {
-		r.Log.Info(fmt.Sprintf("OpenStackMACAddress CR successfully reconciled - operation: %s", string(op)))
+		cond.Message = fmt.Sprintf("OpenStackMACAddress CR successfully reconciled - operation: %s", string(op))
+		cond.Reason = ospdirectorv1beta1.ConditionReason(ospdirectorv1beta1.OsClientCondReasonPodProvisioned)
+		cond.Type = ospdirectorv1beta1.ConditionType(ospdirectorv1beta1.CommonCondTypeProvisioned)
+	}
+
+	return nil
+}
+
+//
+// create or get hash of "tripleo-passwords" controlplane.TripleoPasswordSecret secret
+//
+func (r *OpenStackControlPlaneReconciler) createOrGetTripleoPasswords(
+	instance *ospdirectorv1beta1.OpenStackControlPlane,
+	cond *ospdirectorv1beta1.Condition,
+	envVars *map[string]common.EnvSetter,
+) error {
+	//
+	// check if "tripleo-passwords" controlplane.TripleoPasswordSecret secret already exist
+	//
+	_, secretHash, err := common.GetSecret(r, controlplane.TripleoPasswordSecret, instance.Namespace)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+
+			common.LogForObject(
+				r,
+				fmt.Sprintf("Creating password secret: %s", controlplane.TripleoPasswordSecret),
+				instance,
+			)
+
+			pwSecretLabel := common.GetLabels(instance, controlplane.AppLabel, map[string]string{})
+
+			templateParameters := make(map[string]interface{})
+			templateParameters["TripleoPasswords"] = common.GeneratePasswords()
+			pwSecret := []common.Template{
+				{
+					Name:               controlplane.TripleoPasswordSecret,
+					Namespace:          instance.Namespace,
+					Type:               common.TemplateTypeConfig,
+					InstanceType:       instance.Kind,
+					AdditionalTemplate: map[string]string{},
+					Labels:             pwSecretLabel,
+					ConfigOptions:      templateParameters,
+				},
+			}
+
+			err = common.EnsureSecrets(r, instance, pwSecret, envVars)
+			if err != nil {
+				cond.Message = fmt.Sprintf("Error creating TripleoPasswordsSecret %s", controlplane.TripleoPasswordSecret)
+				cond.Reason = ospdirectorv1beta1.ConditionReason(ospdirectorv1beta1.ControlPlaneReasonTripleoPasswordsSecretCreateError)
+				cond.Type = ospdirectorv1beta1.ConditionType(ospdirectorv1beta1.CommonCondTypeError)
+				err = common.WrapErrorForObject(cond.Message, instance, err)
+
+				return err
+			}
+		} else if err != nil {
```

Comment:
Could this just be an `else`, since we already know `err` is non-nil here?

Reply:
indeed